### PR TITLE
Adding Ability to Query Table By ID

### PR DESCRIPTION
**Pull request by davidov541**
Added the ability to query for the desired table by ID, instead of relying on ordering. 

Also added automated tests for current functionality and added functionality.

### CODE REVIEW

File: html_table_parser/parser.py [hunk 1]
```diff
@@ -33,11 +33,17 @@ def __init__(
         self._current_row = []
         self._current_cell = []
         self.tables = []
+        self.named_tables = {}
+        self.current_table_name = ""
```
Comment (schmijos) [inline html_table_parser/parser.py]: Since `name` is also a valid HTML attribute, I'd change naming here to `current_table_id`.

**Review by schmijos — COMMENTED** [review]
Very nice! And thank you very much for the tests!

Two suggestions for improvement:
* Rename `current_table_name` and `name` to `current_table_id` and `id`
* Add a test for nested tables (we should be able to handle this)

One question:
* You checked-in your `.vscode` project configuration. That's not a best practice in Python right?
  @ypid What do you say?

**Comment by ypid** [comment]
> You checked-in your .vscode project configuration. That's not a best practice in Python right?

I agree. The `.vscode` looks very generic. Is there anything special in their? I would not add editor specific dotfiles to the repo. They also don’t need to go into the `.gitignore` file of the project. If your editor creates such files/directories, you should ignore them on your local machine. If there is really a need for editor config, check out https://editorconfig.org/.

**Comment by davidov541** [comment]
@ypid @schmijos : I made the requested changes. 

I apologize for the mistake about the .vscode, that was a silly mistake on my end. I've removed the files and added the folder to .gitignore so it doesn't happen to future contributors.

I also updated the README to have instructions on how to run the tests.

Let me know if there is anything else you'd like to see before pushing this!

**Comment by schmijos** [comment]
@davidov541 I finally took the time to look properly at your changes. Thank you very much! That you apparently like Münchner Dunkel (Lager or Bock?), actually strengthened my trust in you.

I made you collaborator. Feel free to push this library as you like.

**Comment by davidov541** [comment]
@schmijos Thanks! I will definitely keep an eye on the repo for you and take care of anything that comes up if I have the bandwidth.

And for the record, I'm a bock guy, but in Munich a few years back I had plenty of both!

**Pull request merged**